src/libYARP_manager/src/kbase.cpp

```diff
@@ -1625,7 +1625,7 @@ bool KnowledgeBase::constrainSatisfied(Node* node,
         {
             OSTRINGSTREAM msg;
             msg<<"Some resource dependencies of ";
-            msg<<dynamic_cast<Module*>(resource->owner())->getName();
+            msg<< (dynamic_cast<Module*>(resource->owner()) ? dynamic_cast<Module*>(resource->owner())->getName() : "");
```

Comment:
Why this is marked as warning ? `dynamic_cast<Module*>(resource->owner()` is already checked in the outer if